### ISSUE TITLE
Error resolving referenced formulae ( zcl_excel_common=>determine_resulting_formula )

### ISSUE THREAD

**Issue by StefanSchmoecker** [abap2xlsx/abap2xlsx#348]
The following program demonstrates the error
REPORT.
DATA: lv_resulting_formula  TYPE string,
      lv_referenced_formula TYPE string.

lv_referenced_formula = '=IF($G$2=1,R27+U27+Y27+AC27+AG27+AK27+AO27+AS27+AW27+BA27+BE27+BI27,0)+IF($G$2=2,R27+V27+Y27+AC27+AG27+AK27+AO27+AS27+AW27+BA27+BE27+BI27,0)'.
lv_resulting_formula = zcl_excel_common=>determine_resulting_formula( iv_reference_cell     = 'G26'
                                                                      iv_reference_formula  = lv_referenced_formula
                                                                      iv_current_cell       = 'G27' ).
WRITE:/ lv_referenced_formula.
WRITE:/ lv_resulting_formula.

Resulting formula is truncated

**Comment by StefanSchmoecker** [comment]
Reason:  ZCL_EXCEL_COMMON=>SHIFT_FORMULA uses a textfield of length 132 which truncates the result

**Comment by gregorwolf** [comment]
Hi Stefan,
when you patch that problem please add or extend the unit test for ZCL_EXCEL_COMMON=>SHIFT_FORMULA.
Best regards
Gregor

**Comment by StefanSchmoecker** [comment]
Fixed with revision 406 
Gregor: I thought about that - but I can't come up with a good unit test for this.
The problem are long ( longer than 132 characters ) formulae and the problem has not really been solved but only made much less probable. ( now 2000 characters allowed - I was afraid to use a string because I currently use a 7.4 system which allows more string functions then previous releases and don't have the time to understand what exactly happens in that method completly ).

A unit test for this problem would be a test that addresses the length-issue.
Since it is still faulty when using obscenly large formulae I could either create a unit test for a formula that crosses the upper limit.  But this would in turn flag a unit test as incorrect preventing transports if a system is set this way.
Or I could create a unit test for a long long formula that does not exceed the limit - but I see no point in creating a test that only holds because I selected the input such as not to trigger the failure.

If you have any idea what kind of test you would like to see - any suggestions are welcome .
I'll leave this issue open until I receive an answer from you.
